bundles/com.espressif.idf.sdk.config.ui/src/com/espressif/idf/sdk/config/ui/SDKConfigurationEditor.java

```diff
@@ -648,6 +653,40 @@ else if (isVisible && type.equals(IJsonServerConfig.HEX_TYPE))
 				textControl.addModifyListener(addModifyListener(configKey, textControl));
 				addTooltipImage(kConfigMenuItem);
 			}
+			else if (kConfigMenuItem.isMenuConfig())
+			{
+				Button button = new Button(updateUIComposite, SWT.CHECK);
+				button.setText(kConfigMenuItem.getTitle());
+				button.setLayoutData(new GridData(SWT.NONE, SWT.NONE, false, false, 2, 1));
+				button.setToolTipText(helpInfo);
+				if (configValue != null)
+				{
+					button.setSelection((boolean) configValue);
+				}
+				button.addSelectionListener(new SelectionAdapter()
+				{
+					@Override
+					public void widgetSelected(SelectionEvent e)
+					{
+						JSONObject jsonObj = new JSONObject();
+						jsonObj.put(configKey, button.getSelection());
+						executeCommand(jsonObj);
+						renderMenuItems(kConfigMenuItem);
+					}
+
+				});
```

Comment:
As this code is the same for `IJsonServerConfig.BOOL_TYPE`, could you extract to a method and utilize it.